document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -189,6 +189,42 @@ public boolean updateSubDoc(Key key, String subDocPath, Document subDocument) {
     return false;
   }
 
+  @Override
+  public BulkUpdateResult bulkUpdateSubDocs(Map<Key, Map<String, Document>> documents)
+      throws Exception {
+    String updateSubDocSQL =
+        String.format(
+            "UPDATE %s SET %s=jsonb_set(%s, ?::text[], ?::jsonb) WHERE %s = ?",
+            collectionName, DOCUMENT, DOCUMENT, ID);
+    try {
+      PreparedStatement preparedStatement = client.prepareStatement(updateSubDocSQL);
+      for (Key key : documents.keySet()) {
+        Map<String, Document> subDocuments = documents.get(key);
+        for (String subDocPath : subDocuments.keySet()) {
+          Document subDocument = subDocuments.get(subDocPath);
+          String jsonSubDocPath = getJsonSubDocPath(subDocPath);
+          String jsonString = subDocument.toJson();
+          preparedStatement.setString(1, jsonSubDocPath);
+          preparedStatement.setString(2, jsonString);
+          preparedStatement.setString(3, key.toString());
+          preparedStatement.addBatch();
+        }
+      }
+      int[] updateCounts = preparedStatement.executeBatch();
+      if (LOGGER.isDebugEnabled()) {
+        LOGGER.debug("Write result: {}", updateCounts);
+      }
+      int totalUpdateCount = 0;
+      for (int update : updateCounts) {
+        totalUpdateCount += update;
+      }
+      return new BulkUpdateResult(totalUpdateCount);
```

Comment:
Can we just do `return new BulkUpdateResult(updateCounts.length);` and get rid of `totalUpdateCount`?